tests/js/pouchdb-component-tests/pouchdb-component-common-tests.js

```diff
@@ -0,0 +1,354 @@
+// Browser and Node shared fixtures for the `gpii.pouch` PouchDB component tests.
+/* eslint-env node */
+"use strict";
+var fluid  = fluid || require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+var jqUnit = jqUnit || require("node-jqunit");
+
+if (!gpii.pouch) {
+    require("../../../")
+    gpii.pouch.loadTestingSupport();
+}
+
+fluid.registerNamespace("gpii.tests.pouchdb.component.common");
+
+// A simple function to test support for "map" functions using the query method.
+/* globals emit */
+gpii.tests.pouchdb.component.common.map = function (doc) {
+    if (doc.color) {
+        emit(doc.color, doc);
+    }
+};
+
+// A simple function to test the support for "reduce" functions using the query method.
+gpii.tests.pouchdb.component.common.reduce = function (keys, values, rereduce) {
```

Comment:
I added links to the relevant CouchDB documentation and briefly explained the output of both the map and reduce functions.